File: 507-perfect-number/perfect-number.cpp
class Solution {
public:
    bool checkPerfectNumber(int num) {
        if (num == 1)
            return false;
        int answer = 1, i = 2;
        while (i * i < num) {
            if (num % i == 0)
                answer += i + num / i;
            i += 1;
        }
        if (i * i == num)
            answer += i;

        return answer == num;
    }
};

File: 507-perfect-number/perfect-number.py
class Solution(object):
    def checkPerfectNumber(self, num):
        if num == 1:
            return False
        answer = 1
        for i in range(2, (int(sqrt(num)) + 1)):
            if num % i == 0:
                answer += i
                answer += num / i
        if answer == num:
            return True
        return False
        